### ISSUE TITLE
Version 0.9.5 OpenIndiana build fails

### ISSUE THREAD

**Issue by 3eka** [maxbube/mydumper#131]
Hi,
a month ago, I have managed to compile version 0.9.3 w/o problems on my OpenIndiana /hipster distribution...

Now, it fails:
```bash
Cloning into 'mydumper'...
-- The C compiler identification is GNU 6.4.0
-- The CXX compiler identification is GNU 6.4.0
-- Check for working C compiler: /usr/gcc/6/bin/gcc
-- Check for working C compiler: /usr/gcc/6/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/gcc/6/bin/g++
-- Check for working CXX compiler: /usr/gcc/6/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done  
-- Detecting CXX compile features
-- Detecting CXX compile features - done   
-- Using mysql-config: /pz/MySQL/5.7_64/bin/mysql_config
-- Found MySQL: /pz/MySQL/5.7_64/include, /pz/MySQL/5.7_64/lib/libmysqlclient_r.so
-- Found ZLIB: /usr/lib/64/libz.so (found version "1.2.11") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Checking for one of the modules 'glib-2.0'
-- Checking for one of the modules 'gthread-2.0'
-- Checking for module 'libpcre'
--   Found libpcre, version 8.41
1
-- ------------------------------------------------
-- MYSQL_CONFIG = /pz/MySQL/5.7_64/bin/mysql_config
-- CMAKE_INSTALL_PREFIX = /pz/SFW
-- BUILD_DOCS = ON
-- WITH_BINLOG = OFF
-- RUN_CPPCHECK = OFF
-- Change a values with: cmake -D<Variable>=<Value>
-- ------------------------------------------------
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /export/home/3eka/tmp/SFW_build/mydumper/bld
Scanning dependencies of target mydumper
[  8%] Building C object CMakeFiles/mydumper.dir/mydumper.c.o
[ 16%] Building C object CMakeFiles/mydumper.dir/server_detect.c.o
[ 25%] Building C object CMakeFiles/mydumper.dir/g_unix_signal.c.o
[ 33%] Building C object CMakeFiles/mydumper.dir/connection.c.o
[ 41%] Building C object CMakeFiles/mydumper.dir/getPassword.c.o
/export/home/3eka/tmp/SFW_build/mydumper/getPassword.c: In function 'passwordPrompt':
/export/home/3eka/tmp/SFW_build/mydumper/getPassword.c:7:13: error: implicit declaration of function 'getpass' [-Werror=implicit-function-declaration]
  password = getpass("Enter MySQL Password: ");
             ^~~~~~~
/export/home/3eka/tmp/SFW_build/mydumper/getPassword.c:7:11: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
  password = getpass("Enter MySQL Password: ");
           ^
cc1: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/mydumper.dir/build.make:159: CMakeFiles/mydumper.dir/getPassword.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/mydumper.dir/all] Error 2
gmake: *** [Makefile:130: all] Error 2
```

How to fix that?

Best regards. 
BTW, great tool

**Comment by 3eka** [comment]
Managed to compile version 0.9.5 with this shell
```bash
# Remove POSIX_SOURCE definition (so we can use modern GCC)
# sed -i '/^#define _POSIX_SOURCE/d' ../g_unix_signal.c
#  Fix for error: implicit declaration of function 'getpass'
# sed -i $'s|^#include <stdio.h>|#include <stdlib.h>\\\n#include <stdio.h>|' ../getPassword.c

export CC=/usr/gcc/7/bin/gcc
export CXX=/usr/gcc/7/bin/g++

export PREFIX=/pz/SFW
export myARCH=64
export MY_MYSQL=/pz/MySQL/5.7_${myARCH}

export CFLAGS="-lm -m${myARCH} -I${PREFIX}/include -I${MY_MYSQL}/include -Wl,-rpath,${PREFIX}/lib -D_FILE_OFFSET_BITS=64 -D_XOPEN_SOURCE=600 -D__EXTENSIONS__"
export CXXFLAGS="-lm -m${myARCH} -I${PREFIX}/include -I${MY_MYSQL}/include -Wl,-rpath,${PREFIX}/lib"
export LD_CONFIG="-rpath ${PREFIX}/lib:${MY_MYSQL}/lib"
export LDFLAGS="-lm -m${myARCH} -R${PREFIX}/lib -L${PREFIX}/lib -R${MY_MYSQL}/lib -L${MY_MYSQL}/lib -lstdc++"

PKG_CONFIG_PATH="/usr/lib/pkgconfig" \
  cmake .. -DCMAKE_INSTALL_PREFIX=${PREFIX} \
           -DMYSQL_CONFIG=${MY_MYSQL}/bin/mysql_config \
           -DMYSQL_INCLUDE_DIR=${MY_MYSQL}/include \
           -DMYSQL_LIBRARIES=${MY_MYSQL}/lib/libmysqlclient_r.so \
           -DPC_PCRE_INCLUDEDIR=${PREFIX}/include \
           -DPC_PCRE_LIBDIR=${PREFIX}/lib \
           -DPCRE_INCLUDE_DIR=${PREFIX}/include \
           -DPCRE_PCRE_LIBRARY=${PREFIX}/lib/libpcre.so \
           -DPCRE_PCREPOSIX_LIBRARY=${PREFIX}/lib/libpcreposix.so

gmake all
```
Regards.

**Comment by 3eka** [comment]
Patches:
```bash
$ diff -p g_unix_signal.c.0.9.5 g_unix_signal.c
*** g_unix_signal.c.0.9.5       Mon May 21 15:33:13 2018
--- g_unix_signal.c     Thu Sep  6 11:32:32 2018
***************
*** 1,4 ****
- #define _POSIX_SOURCE
  #include <signal.h>
  #include <glib.h>
  
--- 1,3 ----
```
and
```bash
$ diff -p getPassword.c.0.9.5 getPassword.c
*** getPassword.c.0.9.5 Mon May 21 15:33:13 2018
--- getPassword.c       Thu Sep  6 11:22:35 2018
***************
*** 1,5 ****
- #include <stdio.h>
  #include <unistd.h>
  #include "getPassword.h"
  
  char* passwordPrompt(void) {
--- 1,6 ----
  #include <unistd.h>
+ #include <stdlib.h>
+ #include <stdio.h>
  #include "getPassword.h"
  
  char* passwordPrompt(void) {
```

Regards.

**Comment by 3eka** [comment]
Hi all, guess this can be closed, since in 0.10.0. I can see those patches...

**Comment by 3eka** [comment]
Thanks

**Comment by 3eka** [comment]
Seems patch of *getPassword.c* got lost from https://github.com/maxbube/mydumper/archive/refs/tags/v0.11.3.zip
Can you check, please?

**Comment by davidducos** [comment]
@3eka they were never merged on 0.10.0, so your comment from May was not correct. I will add a new commit that I will request you to review

**Comment by 3eka** [comment]
Huh,
then accept my apologies @davidducos  - was sure that i did checked it.
Anyway, thanks for looking to this one.

Regards.
